### ISSUE TITLE
Error when downloading libraries using gem-maven-plugin

### ISSUE THREAD

**Issue by abelsromero** [ProgramFan/asciidoctor-bibtex#4]
The dependant gem `rubysl-json:2.0.2` is no longer available in RubyGems (https://rubygems.org/gems/rubysl-json) and this causes an error when trying to integrate asciidoctor-bibtex into a maven project:
```
rubygems:namae:gem:0.8.2 -> rubygems:rubysl:gem:2.0.8 -> rubygems:rubysl-json:gem:[2.0,2.99999]: No versions available for rubygems:rubysl-json:gem:[2.0,2.99999] within specified range 
```
The issue is originated by the namae:0.8.2 gem, and adding a new versions does not fix the resolution issue.

I'm not a ruby guy, so I'm just wondering if it's possible to update the dependencies of bibtex to newer versions that are available.

Thanks,

**Comment by ProgramFan** [comment]
Thanks, abel. I wil look into this issue. I am curious why rubysl-json comes as a dependency since this gem does not use json explicitly.

**Comment by ProgramFan** [comment]
@abelsromero Hi, Abel, Would you please test if the problem still exists in the master? I have made changes to the version requirements.

**Comment by ProgramFan** [comment]
It looks like maven requires a wrong version of namae gem is used, 0.8.2 is the only version with rubysl dependency. Upgrade gems shall fix the problem.

This is possible a bug in your maven project, which I have no prior experiences. :(

**Comment by abelsromero** [comment]
The thing is that the maven plugin that allows the integration between Java and Maven checks all versions when gem has a range defined. The problem here is that it fails when one dependency is not found, whereas gem command ignores it if it can find a newer is available...I guess.

Looking into it with more detail namae comes from citeproc, so unless you add namae as a direct dependency the only possible fix (using dependencies) is to update citeproc.
I found a workarround in maven which is adding all dependencies manually.
I was hoping to find a solution via your extension, but seems to me we got to a dead end :disappointed:  Thanks a lot for you time.
